### PR TITLE
Add grunt compiler and html2js templates

### PR DESCRIPTION
**Pull request by jmuharsky**
This compiles the .html templates into /client/perfkit_template.js, which will prevent caching in the future and improve performance.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,11 +56,25 @@ Set up your workstation
...
          git submodule update --init
 
+11. Install NodeJS.
+
+         sudo apt-get install nodejs
+
+12. Install the Node Package Manager (NPM) packages for Gulp and dependencies.
+
+         npm install --global gulp
+         npm install --save-dev gulp
```
Comment (cmccoy) [inline README.md]: Consider adding a `package.json`?

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+var gulp = require('gulp');
+
+var closureCompiler = require('gulp-closure-compiler');
```
Comment (cmccoy) [inline gulpfile.js]: Does this need to be added as a dependency in the README? Or is it transitive?

**Comment by cmccoy** [comment]
This is great to see. 2 minor comments, otherwise LGTM.

File: README.md (hunk 1 above)
Comment (jmuharsky) [inline README.md]: Done.

File: gulpfile.js (hunk 2 above)
Comment (jmuharsky) [inline gulpfile.js]: Needed to be added; added to package.json.

**Comment by cmccoy** [comment]
LGTM.

**Pull request merged**